### ISSUE TITLE
One() Finisher Returning nil

### ISSUE THREAD

**Issue by ossie-git** [volatiletech/sqlboiler#462]
**NOTE** I don't know if this is an issue with sqlboiler or sqlboiler-sqlite3

I'm using the Northwind SQLite DB to test sqlboiler: https://github.com/jpwhite3/northwind-SQLite3/raw/master/Northwind_small.sqlite  I'm using the latest version of sqlboiler (SQLBoiler v3.1.0)

When using the One() finalizer, it returns nil for some tables while working fine with others

I generated the models using the following sqlbuilder.toml file:

```
[sqlite3]
dbname = "Northwind_small.sqlite"
```

Here's an example of a query that returns nil.  I enabled debugging so I could verify the query versus what the DB returns if I run the same query directly against the DB:

```
boil.SetDB(db)
boil.DebugMode = true

p, err := models.Products().One(context.Background(), db)
fmt.Println("Product ID:", p)
```

The debug output shows that it executed:

```
SELECT * FROM "Product" LIMIT 1;
```

If I run the above query against the table directly, it returns the following result:

```
$ sqlite3 Northwind_small.sqlite
SQLite version 3.19.3 2017-06-27 16:48:08
Enter ".help" for usage hints.

sqlite> SELECT * FROM "Product" LIMIT 1;
1|Chai|1|1|10 boxes x 20 bags|18|39|0|10|0
```

For some reason, One() works properly for some tables while failing in others, even though all of the tables have entries.

It failed against the following tables: Product, OrderDetail

### Further information. What did you do, what did you expect?

I expected it to return the first entry in the table

**Comment by aarondl** [comment]
In your code it doesn't show you checking the error. If you're new to Go, you always check the error return first before you know it's safe to use the other returned value, be sure to do so everywhere in your code. When I run the same code and check the error I get this:

```
models: failed to execute a one query for Product: failed to bind pointers to obj: sql: Scan error on column index 5, name "UnitPrice": cannot scan decimal value: 18
```

Note that both of your tables that fail (Product, OrderDetail) both have the Decimal type in them.

That column type in the database is:
```
  "UnitPrice" DECIMAL NOT NULL,
```

sqlite3 driver defines the decimal type as `types.Decimal` which seems sane. And this is reflected in the struct that's generated:

```
	UnitPrice       types.Decimal `boil:"UnitPrice" json:"UnitPrice" toml:"UnitPrice" yaml:"UnitPrice"`
```

I wrote a test program that checks to see if I can decode an int64 value of 18 with `types.Decimal` and the answer is no:

```go
package main

import (
	"fmt"

	"github.com/volatiletech/sqlboiler/types"
)

func main() {
	var a types.Decimal

	fmt.Println(a.Scan(int64(18)))
}
```

Output is:

```text
cannot scan decimal value: 18
```

This was not being handled in the `types.Decimal` scan. I've added a case for it in the dev branch. I'm able to use the dev branch to run your sample code successfully.

So this should be resolved there and will be released at a later date. You can use the dev branch in the interim. We just released `v3.2.0` recently fwiw. So it will probably be a bit before this sees release. Thanks for submitting the issue.

**Comment by ossie-git** [comment]
Thanks a lot
